src/Chapter3.hs

```diff
@@ -971,6 +1146,39 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+days = ["Sun", "Mon","Tue" ,"Wed", "Thu", "Fri", "Sat"]
+
+data WeekDay = WeekDay
+  {
+    day :: String
+  }
+
+isWeekend :: String -> Bool
+isWeekend "Sun" = True
+isWeekend "Sat" = True
+isWeekend _ = False
+
+-- Brute-forcing this
+nextDay :: String -> String
+nextDay "Sun" = "Mon"
+nextDay "Mon" = "Tue"
+nextDay "Tue" = "Wed"
+nextDay "Wed" = "Thu"
+nextDay "Thu" = "Fri"
+nextDay "Fri" = "Sat"
+nextDay "Sat" = "Sun"
+
+daysToParty :: String -> Int
+daysToParty "Fri" = 0
+daysToParty "Sat" = 0
+daysToParty "Sun" = 0
+daysToParty "Mon" = 4
+daysToParty "Tue" = 3
+daysToParty "Wed" = 2
+daysToParty "Thu" = 1
```

Comment:
If you change the type to the sum type, you would be able to use `Enum` typeclass instance to implement this function easily and safely 🙂